toolchain/check/check_unit.cpp

```diff
@@ -394,6 +396,19 @@ auto CheckUnit::ProcessNodeIds() -> bool {
   return true;
 }
 
+auto CheckUnit::CheckRequiredDeclarations() -> void {
+  for (const auto& function : context_.functions().array_ref()) {
+    if (!function.first_owning_decl_id.has_value() &&
+        function.extern_library_id == context_.sem_ir().library_id()) {
+      // TODO: add condition function.package == context_.sem_ir().package_id()
+      CARBON_DIAGNOSTIC(MissingDeclarationInApi, Error,
+                        "no owning function declaration or definition found "
+                        "for declaration in non-owning file");
```

Comment:
```suggestion
                        "owning declaration required for non-owning declaration");
```

Phrasing suggestion, trying to tinker to be more succinct. Note we talk about definitions as a subtype of declaration; either a forward declaration or definition would be the precise terms we'd use, but both are "owning declarations".

Also, suggesting to drop "function" because we'll want to use this same diagnostic for other entities. We only have non-owning functions right now, but we should end up with similar for classes and so on.